src/directions.js

```diff
@@ -15,6 +17,36 @@ var Directions = L.Class.extend({
         this._waypoints = [];
     },
 
+    geocode: function(waypoint, proximity, cb) {
+        this._geocode = corslite(L.Util.template(Directions.GEOCODER_TEMPLATE, {
```

Comment:
Multiple geocoding requests are being made in parallel. When aborting, we need to abort them all. So: `this._requests.push(corslite(...))`, and then the abort is like:

```
this._requests.forEach(function (request) { request.abort(); });
this._requests = [];
```